Quota exceeded issues shouldn't be handled as FORBIDDEN (403)

Quota exceeded issues shouldn't be handled as FORBIDDEN (403)

Currently when GCE operation fails with "quote limit exceeded" error, it contains 403 error code and as a result handled as FORBIDDEN:

``` json
{ "kind": "compute#operation", "id": "604106744976298432", "name": "operation-1413455311761-50587b5c36e68-ca2bfdd4-4d854ce2", "operationType": "insert", "targetLink": "https://clients6.google.com/compute/v1/projects/grid-impl-1206/global/networks/gce-test-case-grid-gce-2014-10-16-14-28", "status": "DONE", "user": "25777443288-lt1oeoag4b49v87b9nef06hmjprdhc9s@developer.gserviceaccount.com", "progress": 100, "insertTime": "2014-10-16T03:28:32.009-07:00", "startTime": "2014-10-16T03:28:32.162-07:00", "endTime": "2014-10-16T03:28:32.574-07:00", "error": { "errors": [ { "code": "QUOTA_EXCEEDED", "message": "Quota 'NETWORKS' exceeded. Limit: 25.0" } ] }, "httpErrorStatusCode": 403, "httpErrorMessage": "FORBIDDEN", "selfLink": "https://www.googleapis.com/compute/v1/projects/grid-impl-1206/global/operations/operation-1413455311761-50587b5c36e68-ca2bfdd4-4d854ce2" }
```

Handle "QUOTA_EXCEEDED" error code as properly. Add mapping between error code and HTTP status:

```
03:20:48.254 [DEBUG] [TestEventLogger]     03:20:48.254 [Test worker] ERROR c.w.g.e.mappers.CloudExceptionMapper - Error handling a request: 6f61222898ff9d7d
03:20:48.255 [DEBUG] [TestEventLogger]     org.dasein.cloud.google.common.GoogleOperationFailedException: UNKNOWN operation [insert] failed for [gce-test-case-grid-gce-2014-10-16-03-13] with error: "Quota 'NETWORKS' exceeded.  Limit: 25.0"
03:20:48.255 [DEBUG] [TestEventLogger]      at org.dasein.cloud.google.common.GoogleOperationFailedException.create(GoogleOperationFailedException.java:35) ~[dasein-cloud-google-2015.01.1-SNAPSHOT.jar:na]
03:20:48.256 [DEBUG] [TestEventLogger]      at org.dasein.cloud.google.util.model.GoogleOperations.handleOperationStatus(GoogleOperations.java:136) ~[dasein-cloud-google-2015.01.1-SNAPSHOT.jar:na]
03:20:48.256 [DEBUG] [TestEventLogger]      at org.dasein.cloud.google.util.model.GoogleOperations.logOperationStatusOrFail(GoogleOperations.java:98) ~[dasein-cloud-google-2015.01.1-SNAPSHOT.jar:na]
03:20:48.257 [DEBUG] [TestEventLogger]      at org.dasein.cloud.google.compute.server.GoogleOperationSupport.getGlobalOperation(GoogleOperationSupport.java:239) ~[dasein-cloud-google-2015.01.1-SNAPSHOT.jar:na]
03:20:48.257 [DEBUG] [TestEventLogger]      at org.dasein.cloud.google.compute.server.GoogleOperationSupport.getUpdatedOperation(GoogleOperationSupport.java:129) ~[dasein-cloud-google-2015.01.1-SNAPSHOT.jar:na]
03:20:48.257 [DEBUG] [TestEventLogger]      at org.dasein.cloud.google.compute.server.GoogleOperationSupport$1.call(GoogleOperationSupport.java:107) ~[dasein-cloud-google-2015.01.1-SNAPSHOT.jar:na]
03:20:48.258 [DEBUG] [TestEventLogger]      at org.dasein.cloud.google.compute.server.GoogleOperationSupport$1.call(GoogleOperationSupport.java:101) ~[dasein-cloud-google-2015.01.1-SNAPSHOT.jar:na]
03:20:48.258 [DEBUG] [TestEventLogger]      at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:334) ~[na:1.7.0_25]
03:20:48.259 [DEBUG] [TestEventLogger]      at java.util.concurrent.FutureTask.run(FutureTask.java:166) ~[na:1.7.0_25]
03:20:48.259 [DEBUG] [TestEventLogger]      at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145) ~[na:1.7.0_25]
03:20:48.259 [DEBUG] [TestEventLogger]      at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615) ~[na:1.7.0_25]
03:20:48.260 [DEBUG] [TestEventLogger]      at java.lang.Thread.run(Thread.java:724) ~[na:1.7.0_25]
03:20:48.260 [DEBUG] [TestEventLogger] 
03:20:48.260 [DEBUG] [TestEventLogger] Gradle test STANDARD_ERROR
03:20:48.260 [DEBUG] [TestEventLogger]     Oct 16, 2014 3:20:48 AM org.glassfish.jersey.filter.LoggingFilter log
03:20:48.261 [DEBUG] [TestEventLogger]     INFO: 166 * Server responded with a response on thread Test worker
03:20:48.261 [DEBUG] [TestEventLogger]     166 < 403
03:20:48.262 [DEBUG] [TestEventLogger]     166 < Content-Type: application/json
03:20:48.262 [DEBUG] [TestEventLogger] 
03:20:48.262 [DEBUG] [TestEventLogger]     Oct 16, 2014 3:20:48 AM org.glassfish.jersey.filter.LoggingFilter log
03:20:48.263 [DEBUG] [TestEventLogger]     INFO: 332 * Client response received on thread Test worker
03:20:48.263 [DEBUG] [TestEventLogger]     332 < 403
03:20:48.264 [DEBUG] [TestEventLogger]     332 < Content-Length: 74
03:20:48.264 [DEBUG] [TestEventLogger]     332 < Content-Type: application/json
03:20:48.264 [DEBUG] [TestEventLogger]     Unauthorized operation - You are not authorized to perform that operation.
03:20:48.265 [DEBUG] [TestEventLogger] 
03:20:48.273 [DEBUG] [TestEventLogger] 
03:20:48.273 [DEBUG] [TestEventLogger] Gradle test STANDARD_OUT
03:20:48.274 [DEBUG] [TestEventLogger]     03:20:48.272 [Test worker] ERROR com.weather.grid.client.BaseClient - Unexpected error...
03:20:48.274 [DEBUG] [TestEventLogger]     com.weather.grid.model.GridException: Unauthorized operation - You are not authorized to perform that operation.
03:20:48.274 [DEBUG] [TestEventLogger]      at sun.reflect.GeneratedConstructorAccessor148.newInstance(Unknown Source) ~[na:na]
03:20:48.275 [DEBUG] [TestEventLogger]      at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[na:1.7.0_25]
03:20:48.275 [DEBUG] [TestEventLogger]      at java.lang.reflect.Constructor.newInstance(Constructor.java:526) ~[na:1.7.0_25]
03:20:48.276 [DEBUG] [TestEventLogger]      at org.codehaus.groovy.reflection.CachedConstructor.invoke(CachedConstructor.java:77) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.276 [DEBUG] [TestEventLogger]      at org.codehaus.groovy.runtime.callsite.ConstructorSite$ConstructorSiteNoUnwrapNoCoerce.callConstructor(ConstructorSite.java:102) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.276 [DEBUG] [TestEventLogger]      at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallConstructor(CallSiteArray.java:57) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.277 [DEBUG] [TestEventLogger]      at org.codehaus.groovy.runtime.callsite.ConstructorSite$ConstructorSiteNoUnwrapNoCoerce.callConstructor(ConstructorSite.java:107) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.277 [DEBUG] [TestEventLogger]      at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callConstructor(AbstractCallSite.java:198) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.278 [DEBUG] [TestEventLogger]      at com.weather.grid.client.BaseClient.handleUniformInterfaceException(BaseClient.groovy:333) ~[test/:na]
03:20:48.278 [DEBUG] [TestEventLogger]      at sun.reflect.GeneratedMethodAccessor400.invoke(Unknown Source) ~[na:na]
03:20:48.279 [DEBUG] [TestEventLogger]      at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.7.0_25]
03:20:48.279 [DEBUG] [TestEventLogger]      at java.lang.reflect.Method.invoke(Method.java:606) ~[na:1.7.0_25]
03:20:48.279 [DEBUG] [TestEventLogger]      at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:90) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.280 [DEBUG] [TestEventLogger]      at org.codehaus.groovy.runtime.callsite.StaticMetaMethodSite$StaticMetaMethodSiteNoUnwrapNoCoerce.invoke(StaticMetaMethodSite.java:148) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.280 [DEBUG] [TestEventLogger]      at org.codehaus.groovy.runtime.callsite.StaticMetaMethodSite.callStatic(StaticMetaMethodSite.java:99) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.281 [DEBUG] [TestEventLogger]      at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallStatic(CallSiteArray.java:53) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.281 [DEBUG] [TestEventLogger]      at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callStatic(AbstractCallSite.java:157) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.282 [DEBUG] [TestEventLogger]      at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callStatic(AbstractCallSite.java:165) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.282 [DEBUG] [TestEventLogger]      at com.weather.grid.client.BaseClient.post(BaseClient.groovy:160) ~[test/:na]
03:20:48.282 [DEBUG] [TestEventLogger]      at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.7.0_25]
03:20:48.283 [DEBUG] [TestEventLogger]      at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57) ~[na:1.7.0_25]
03:20:48.283 [DEBUG] [TestEventLogger]      at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.7.0_25]
03:20:48.284 [DEBUG] [TestEventLogger]      at java.lang.reflect.Method.invoke(Method.java:606) ~[na:1.7.0_25]
03:20:48.284 [DEBUG] [TestEventLogger]      at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoCachedMethodSiteNoUnwrapNoCoerce.invoke(PogoMetaMethodSite.java:272) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.285 [DEBUG] [TestEventLogger]      at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.callCurrent(PogoMetaMethodSite.java:52) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.286 [DEBUG] [TestEventLogger]      at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallCurrent(CallSiteArray.java:49) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.286 [DEBUG] [TestEventLogger]      at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:133) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.286 [DEBUG] [TestEventLogger]      at com.weather.grid.client.BaseClient.post(BaseClient.groovy:107) ~[test/:na]
03:20:48.286 [DEBUG] [TestEventLogger]      at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.7.0_25]
03:20:48.286 [DEBUG] [TestEventLogger]      at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57) ~[na:1.7.0_25]
03:20:48.287 [DEBUG] [TestEventLogger]      at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.7.0_25]
03:20:48.287 [DEBUG] [TestEventLogger]      at java.lang.reflect.Method.invoke(Method.java:606) ~[na:1.7.0_25]
03:20:48.288 [DEBUG] [TestEventLogger]      at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoCachedMethodSiteNoUnwrapNoCoerce.invoke(PogoMetaMethodSite.java:272) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.289 [DEBUG] [TestEventLogger]      at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.callCurrent(PogoMetaMethodSite.java:52) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.289 [DEBUG] [TestEventLogger]      at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallCurrent(CallSiteArray.java:49) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.289 [DEBUG] [TestEventLogger]      at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:133) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.289 [DEBUG] [TestEventLogger]      at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:153) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.289 [DEBUG] [TestEventLogger]      at com.weather.grid.client.BaseClient.post(BaseClient.groovy:95) ~[test/:na]
03:20:48.290 [DEBUG] [TestEventLogger]      at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.7.0_25]
03:20:48.290 [DEBUG] [TestEventLogger]      at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57) ~[na:1.7.0_25]
03:20:48.291 [DEBUG] [TestEventLogger]      at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.7.0_25]
03:20:48.291 [DEBUG] [TestEventLogger]      at java.lang.reflect.Method.invoke(Method.java:606) ~[na:1.7.0_25]
03:20:48.292 [DEBUG] [TestEventLogger]      at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:90) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.292 [DEBUG] [TestEventLogger]      at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:233) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.292 [DEBUG] [TestEventLogger]      at org.codehaus.groovy.runtime.metaclass.ClosureMetaClass.invokeMethod(ClosureMetaClass.java:361) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.293 [DEBUG] [TestEventLogger]      at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:906) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.293 [DEBUG] [TestEventLogger]      at org.codehaus.groovy.runtime.callsite.PogoMetaClassSite.callCurrent(PogoMetaClassSite.java:66) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.294 [DEBUG] [TestEventLogger]      at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallCurrent(CallSiteArray.java:49) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.294 [DEBUG] [TestEventLogger]      at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:133) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.294 [DEBUG] [TestEventLogger]      at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:153) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.295 [DEBUG] [TestEventLogger]      at com.weather.grid.client.network.NetworkResourceClient$_createNetwork_closure1.doCall(NetworkResourceClient.groovy:39) ~[test/:na]
03:20:48.295 [DEBUG] [TestEventLogger]      at com.weather.grid.client.network.NetworkResourceClient$_createNetwork_closure1.doCall(NetworkResourceClient.groovy) ~[test/:na]
03:20:48.295 [DEBUG] [TestEventLogger]      at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.7.0_25]
03:20:48.296 [DEBUG] [TestEventLogger]      at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57) ~[na:1.7.0_25]
03:20:48.296 [DEBUG] [TestEventLogger]      at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.7.0_25]
03:20:48.296 [DEBUG] [TestEventLogger]      at java.lang.reflect.Method.invoke(Method.java:606) ~[na:1.7.0_25]
03:20:48.297 [DEBUG] [TestEventLogger]      at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:90) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.297 [DEBUG] [TestEventLogger]      at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:233) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.297 [DEBUG] [TestEventLogger]      at org.codehaus.groovy.runtime.metaclass.ClosureMetaClass.invokeMethod(ClosureMetaClass.java:272) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.298 [DEBUG] [TestEventLogger]      at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:906) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.298 [DEBUG] [TestEventLogger]      at org.codehaus.groovy.runtime.callsite.PogoMetaClassSite.call(PogoMetaClassSite.java:39) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.298 [DEBUG] [TestEventLogger]      at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:45) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.298 [DEBUG] [TestEventLogger]      at org.codehaus.groovy.runtime.callsite.PogoMetaClassSite.call(PogoMetaClassSite.java:54) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.299 [DEBUG] [TestEventLogger]      at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:112) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.299 [DEBUG] [TestEventLogger]      at com.weather.grid.client.BaseClient.handleClientException(BaseClient.groovy:307) ~[test/:na]
03:20:48.299 [DEBUG] [TestEventLogger]      at sun.reflect.GeneratedMethodAccessor403.invoke(Unknown Source) ~[na:na]
03:20:48.300 [DEBUG] [TestEventLogger]      at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.7.0_25]
03:20:48.300 [DEBUG] [TestEventLogger]      at java.lang.reflect.Method.invoke(Method.java:606) ~[na:1.7.0_25]
03:20:48.300 [DEBUG] [TestEventLogger]      at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:90) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.300 [DEBUG] [TestEventLogger]      at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:233) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.301 [DEBUG] [TestEventLogger]      at groovy.lang.MetaClassImpl.invokeStaticMethod(MetaClassImpl.java:1330) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.301 [DEBUG] [TestEventLogger]      at org.codehaus.groovy.runtime.callsite.StaticMetaClassSite.callStatic(StaticMetaClassSite.java:62) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.301 [DEBUG] [TestEventLogger]      at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallStatic(CallSiteArray.java:53) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.302 [DEBUG] [TestEventLogger]      at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callStatic(AbstractCallSite.java:157) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.302 [DEBUG] [TestEventLogger]      at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callStatic(AbstractCallSite.java:165) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.302 [DEBUG] [TestEventLogger]      at com.weather.grid.client.network.NetworkResourceClient.createNetwork(NetworkResourceClient.groovy:42) ~[test/:na]
03:20:48.302 [DEBUG] [TestEventLogger]      at com.weather.grid.services.network.NetworkService$createNetwork.call(Unknown Source) ~[na:na]
03:20:48.303 [DEBUG] [TestEventLogger]      at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:45) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.303 [DEBUG] [TestEventLogger]      at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:108) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.303 [DEBUG] [TestEventLogger]      at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:116) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.304 [DEBUG] [TestEventLogger]      at com.weather.grid.tests.data.NetworkTestDataProvider.createNetwork(NetworkTestDataProvider.groovy:45) ~[grid-test-suite-2014-10-16-d99667f.jar:na]
03:20:48.304 [DEBUG] [TestEventLogger]      at com.weather.grid.tests.data.NetworkTestDataProvider$createNetwork.callCurrent(Unknown Source) ~[na:na]
03:20:48.304 [DEBUG] [TestEventLogger]      at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallCurrent(CallSiteArray.java:49) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.304 [DEBUG] [TestEventLogger]      at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:133) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.305 [DEBUG] [TestEventLogger]      at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:141) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.305 [DEBUG] [TestEventLogger]      at com.weather.grid.tests.data.NetworkTestDataProvider.prepareTestData(NetworkTestDataProvider.groovy:18) ~[grid-test-suite-2014-10-16-d99667f.jar:na]
03:20:48.305 [DEBUG] [TestEventLogger]      at com.weather.grid.tests.data.TestDataProvider$prepareTestData.call(Unknown Source) ~[na:na]
03:20:48.306 [DEBUG] [TestEventLogger]      at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:45) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.306 [DEBUG] [TestEventLogger]      at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:108) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.306 [DEBUG] [TestEventLogger]      at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:116) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.307 [DEBUG] [TestEventLogger]      at com.weather.grid.tests.data.CachedDataProvider$_closure1.doCall(CachedDataProvider.groovy:27) ~[grid-test-suite-2014-10-16-d99667f.jar:na]
03:20:48.307 [DEBUG] [TestEventLogger]      at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.7.0_25]
03:20:48.307 [DEBUG] [TestEventLogger]      at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57) ~[na:1.7.0_25]
03:20:48.307 [DEBUG] [TestEventLogger]      at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.7.0_25]
03:20:48.308 [DEBUG] [TestEventLogger]      at java.lang.reflect.Method.invoke(Method.java:606) ~[na:1.7.0_25]
03:20:48.308 [DEBUG] [TestEventLogger]      at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:90) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.308 [DEBUG] [TestEventLogger]      at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:233) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.309 [DEBUG] [TestEventLogger]      at org.codehaus.groovy.runtime.metaclass.ClosureMetaClass.invokeMethod(ClosureMetaClass.java:272) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.309 [DEBUG] [TestEventLogger]      at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:906) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.309 [DEBUG] [TestEventLogger]      at groovy.lang.Closure.call(Closure.java:415) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.310 [DEBUG] [TestEventLogger]      at org.codehaus.groovy.runtime.memoize.Memoize$MemoizeFunction.call(Memoize.java:127) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.310 [DEBUG] [TestEventLogger]      at groovy.lang.Closure.call(Closure.java:428) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.310 [DEBUG] [TestEventLogger]      at groovy.lang.Closure$call.call(Unknown Source) ~[na:na]
03:20:48.311 [DEBUG] [TestEventLogger]      at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:45) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.311 [DEBUG] [TestEventLogger]      at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:108) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.311 [DEBUG] [TestEventLogger]      at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:116) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.311 [DEBUG] [TestEventLogger]      at com.weather.grid.tests.data.CachedDataProvider.prepareTestData(CachedDataProvider.groovy:33) ~[grid-test-suite-2014-10-16-d99667f.jar:na]
03:20:48.312 [DEBUG] [TestEventLogger]      at com.weather.grid.tests.data.TestDataProvider$prepareTestData.call(Unknown Source) ~[na:na]
03:20:48.312 [DEBUG] [TestEventLogger]      at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:45) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.312 [DEBUG] [TestEventLogger]      at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:108) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.312 [DEBUG] [TestEventLogger]      at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:116) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.313 [DEBUG] [TestEventLogger]      at com.weather.grid.tests.CloudTestSuiteContext.demandEntityId(CloudTestSuiteContext.groovy:38) ~[grid-test-suite-2014-10-16-d99667f.jar:na]
03:20:48.313 [DEBUG] [TestEventLogger]      at com.weather.grid.tests.CloudTestSuiteContext$demandEntityId.call(Unknown Source) ~[na:na]
03:20:48.313 [DEBUG] [TestEventLogger]      at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:45) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.314 [DEBUG] [TestEventLogger]      at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:108) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.314 [DEBUG] [TestEventLogger]      at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:116) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.314 [DEBUG] [TestEventLogger]      at com.weather.grid.tests.BaseTest.demandEntity(BaseTest.groovy:274) ~[grid-test-suite-2014-10-16-d99667f.jar:na]
03:20:48.314 [DEBUG] [TestEventLogger]      at com.weather.grid.tests.BaseTest$demandEntity.callCurrent(Unknown Source) ~[na:na]
03:20:48.315 [DEBUG] [TestEventLogger]      at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallCurrent(CallSiteArray.java:49) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.315 [DEBUG] [TestEventLogger]      at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:133) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.315 [DEBUG] [TestEventLogger]      at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:145) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.315 [DEBUG] [TestEventLogger]      at com.weather.grid.tests.BaseTest.getNetworkId(BaseTest.groovy:320) ~[grid-test-suite-2014-10-16-d99667f.jar:na]
03:20:48.316 [DEBUG] [TestEventLogger]      at com.weather.grid.tests.BaseTest$getNetworkId.callCurrent(Unknown Source) ~[na:na]
03:20:48.316 [DEBUG] [TestEventLogger]      at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallCurrent(CallSiteArray.java:49) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.316 [DEBUG] [TestEventLogger]      at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:133) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.316 [DEBUG] [TestEventLogger]      at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:137) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.317 [DEBUG] [TestEventLogger]      at com.weather.grid.tests.network.firewall.FirewallServiceTests.setup(FirewallServiceTests.groovy:59) ~[grid-test-suite-2014-10-16-d99667f.jar:na]
03:20:48.317 [DEBUG] [TestEventLogger]      at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.7.0_25]
03:20:48.317 [DEBUG] [TestEventLogger]      at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57) ~[na:1.7.0_25]
03:20:48.318 [DEBUG] [TestEventLogger]      at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.7.0_25]
03:20:48.318 [DEBUG] [TestEventLogger]      at java.lang.reflect.Method.invoke(Method.java:606) ~[na:1.7.0_25]
03:20:48.318 [DEBUG] [TestEventLogger]      at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:84) ~[testng-6.8.5.jar:na]
03:20:48.318 [DEBUG] [TestEventLogger]      at org.testng.internal.Invoker.invokeConfigurationMethod(Invoker.java:564) ~[testng-6.8.5.jar:na]
03:20:48.319 [DEBUG] [TestEventLogger]      at org.testng.internal.Invoker.invokeConfigurations(Invoker.java:213) ~[testng-6.8.5.jar:na]
03:20:48.319 [DEBUG] [TestEventLogger]      at org.testng.internal.Invoker.invokeConfigurations(Invoker.java:138) ~[testng-6.8.5.jar:na]
03:20:48.319 [DEBUG] [TestEventLogger]      at org.testng.internal.TestMethodWorker.invokeBeforeClassMethods(TestMethodWorker.java:175) ~[testng-6.8.5.jar:na]
03:20:48.319 [DEBUG] [TestEventLogger]      at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:107) ~[testng-6.8.5.jar:na]
03:20:48.320 [DEBUG] [TestEventLogger]      at org.testng.TestRunner.privateRun(TestRunner.java:767) ~[testng-6.8.5.jar:na]
03:20:48.320 [DEBUG] [TestEventLogger]      at org.testng.TestRunner.run(TestRunner.java:617) ~[testng-6.8.5.jar:na]
03:20:48.320 [DEBUG] [TestEventLogger]      at org.testng.SuiteRunner.runTest(SuiteRunner.java:334) ~[testng-6.8.5.jar:na]
03:20:48.320 [DEBUG] [TestEventLogger]      at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:329) ~[testng-6.8.5.jar:na]
03:20:48.321 [DEBUG] [TestEventLogger]      at org.testng.SuiteRunner.privateRun(SuiteRunner.java:291) ~[testng-6.8.5.jar:na]
03:20:48.321 [DEBUG] [TestEventLogger]      at org.testng.SuiteRunner.run(SuiteRunner.java:240) ~[testng-6.8.5.jar:na]
03:20:48.321 [DEBUG] [TestEventLogger]      at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52) ~[testng-6.8.5.jar:na]
03:20:48.322 [DEBUG] [TestEventLogger]      at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86) ~[testng-6.8.5.jar:na]
03:20:48.322 [DEBUG] [TestEventLogger]      at org.testng.TestNG.runSuitesSequentially(TestNG.java:1224) ~[testng-6.8.5.jar:na]
03:20:48.322 [DEBUG] [TestEventLogger]      at org.testng.TestNG.runSuitesLocally(TestNG.java:1149) ~[testng-6.8.5.jar:na]
03:20:48.322 [DEBUG] [TestEventLogger]      at org.testng.TestNG.run(TestNG.java:1057) ~[testng-6.8.5.jar:na]
03:20:48.323 [DEBUG] [TestEventLogger]      at org.gradle.api.internal.tasks.testing.testng.TestNGTestClassProcessor.stop(TestNGTestClassProcessor.java:112) ~[na:na]
03:20:48.323 [DEBUG] [TestEventLogger]      at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.stop(SuiteTestClassProcessor.java:58) ~[na:na]
03:20:48.323 [DEBUG] [TestEventLogger]      at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.7.0_25]
03:20:48.323 [DEBUG] [TestEventLogger]      at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57) ~[na:1.7.0_25]
03:20:48.324 [DEBUG] [TestEventLogger]      at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.7.0_25]
03:20:48.324 [DEBUG] [TestEventLogger]      at java.lang.reflect.Method.invoke(Method.java:606) ~[na:1.7.0_25]
03:20:48.324 [DEBUG] [TestEventLogger]      at org.gradle.messaging.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:35) ~[na:na]
03:20:48.325 [DEBUG] [TestEventLogger]      at org.gradle.messaging.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24) ~[na:na]
03:20:48.325 [DEBUG] [TestEventLogger]      at org.gradle.messaging.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:32) ~[na:na]
03:20:48.325 [DEBUG] [TestEventLogger]      at org.gradle.messaging.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:93) ~[na:na]
03:20:48.326 [DEBUG] [TestEventLogger]      at com.sun.proxy.$Proxy2.stop(Unknown Source) ~[na:na]
03:20:48.326 [DEBUG] [TestEventLogger]      at org.gradle.api.internal.tasks.testing.worker.TestWorker.stop(TestWorker.java:113) ~[na:na]
03:20:48.327 [DEBUG] [TestEventLogger]      at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.7.0_25]
03:20:48.327 [DEBUG] [TestEventLogger]      at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57) ~[na:1.7.0_25]
03:20:48.327 [DEBUG] [TestEventLogger]      at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.7.0_25]
03:20:48.327 [DEBUG] [TestEventLogger]      at java.lang.reflect.Method.invoke(Method.java:606) ~[na:1.7.0_25]
03:20:48.328 [DEBUG] [TestEventLogger]      at org.gradle.messaging.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:35) ~[na:na]
03:20:48.328 [DEBUG] [TestEventLogger]      at org.gradle.messaging.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24) ~[na:na]
03:20:48.328 [DEBUG] [TestEventLogger]      at org.gradle.messaging.remote.internal.hub.MessageHub$Handler.run(MessageHub.java:355) ~[na:na]
03:20:48.329 [DEBUG] [TestEventLogger]      at org.gradle.internal.concurrent.DefaultExecutorFactory$StoppableExecutorImpl$1.run(DefaultExecutorFactory.java:66) ~[na:na]
03:20:48.329 [DEBUG] [TestEventLogger]      at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145) ~[na:1.7.0_25]
03:20:48.329 [DEBUG] [TestEventLogger]      at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615) ~[na:1.7.0_25]
03:20:48.329 [DEBUG] [TestEventLogger]      at java.lang.Thread.run(Thread.java:724) ~[na:1.7.0_25]
03:20:48.330 [DEBUG] [TestEventLogger]     Caused by: javax.ws.rs.ForbiddenException: HTTP 403 Forbidden
03:20:48.330 [DEBUG] [TestEventLogger]      at org.glassfish.jersey.client.JerseyInvocation.convertToException(JerseyInvocation.java:953) ~[jersey-client-2.13-SNAPSHOT.jar:na]
03:20:48.330 [DEBUG] [TestEventLogger]      at org.glassfish.jersey.client.JerseyInvocation.translate(JerseyInvocation.java:795) ~[jersey-client-2.13-SNAPSHOT.jar:na]
03:20:48.331 [DEBUG] [TestEventLogger]      at org.glassfish.jersey.client.JerseyInvocation.access$500(JerseyInvocation.java:91) ~[jersey-client-2.13-SNAPSHOT.jar:na]
03:20:48.331 [DEBUG] [TestEventLogger]      at org.glassfish.jersey.client.JerseyInvocation$2.call(JerseyInvocation.java:683) ~[jersey-client-2.13-SNAPSHOT.jar:na]
03:20:48.331 [DEBUG] [TestEventLogger]      at org.glassfish.jersey.internal.Errors.process(Errors.java:315) ~[jersey-common-2.13-SNAPSHOT.jar:na]
03:20:48.331 [DEBUG] [TestEventLogger]      at org.glassfish.jersey.internal.Errors.process(Errors.java:297) ~[jersey-common-2.13-SNAPSHOT.jar:na]
03:20:48.332 [DEBUG] [TestEventLogger]      at org.glassfish.jersey.internal.Errors.process(Errors.java:228) ~[jersey-common-2.13-SNAPSHOT.jar:na]
03:20:48.332 [DEBUG] [TestEventLogger]      at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:424) ~[jersey-common-2.13-SNAPSHOT.jar:na]
03:20:48.332 [DEBUG] [TestEventLogger]      at org.glassfish.jersey.client.JerseyInvocation.invoke(JerseyInvocation.java:679) ~[jersey-client-2.13-SNAPSHOT.jar:na]
03:20:48.333 [DEBUG] [TestEventLogger]      at org.glassfish.jersey.client.JerseyInvocation$Builder.method(JerseyInvocation.java:435) ~[jersey-client-2.13-SNAPSHOT.jar:na]
03:20:48.333 [DEBUG] [TestEventLogger]      at org.glassfish.jersey.client.JerseyInvocation$Builder.post(JerseyInvocation.java:338) ~[jersey-client-2.13-SNAPSHOT.jar:na]
03:20:48.333 [DEBUG] [TestEventLogger]      at javax.ws.rs.client.SyncInvoker$post.call(Unknown Source) ~[na:na]
03:20:48.334 [DEBUG] [TestEventLogger]      at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:45) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.334 [DEBUG] [TestEventLogger]      at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:108) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.334 [DEBUG] [TestEventLogger]      at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:120) ~[groovy-all-2.0.5.jar:2.0.5]
03:20:48.334 [DEBUG] [TestEventLogger]      at com.weather.grid.client.BaseClient.post(BaseClient.groovy:157) ~[test/:na]
03:20:48.335 [DEBUG] [TestEventLogger]      ... 152 common frames omitted
```

--- a/src/main/java/org/dasein/cloud/google/common/GoogleOperationFailedException.java
+++ b/src/main/java/org/dasein/cloud/google/common/GoogleOperationFailedException.java
@@ -1,6 +1,7 @@
 package org.dasein.cloud.google.common;
 
 import com.google.api.services.compute.model.Operation;
+import com.google.common.collect.Maps;
 import org.dasein.cloud.CloudErrorType;
 import org.dasein.cloud.CloudException;
 
@@ -8,6 +9,9 @@ import javax.annotation.Nonnegative;
 import javax.annotation.Nonnull;
 import javax.annotation.Nullable;
 import java.util.List;
+import java.util.Map;
+
+import static com.google.api.services.compute.model.Operation.Error.Errors;
 
 /**
  * Exceptions to be using when an operation fails on GCE side
@@ -16,22 +20,36 @@ import java.util.List;
  * @since 01.05.2014
  */
 public class GoogleOperationFailedException extends CloudException {
+
 	private static final long serialVersionUID = -4100288452798844941L;
 
-	public GoogleOperationFailedException(@Nonnegative int httpCode, @Nullable String providerCode, @Nonnull String msg) {
+	private static final Map<String, Integer> OPERATION_CODES = Maps.newHashMap();
+
+	static {
+		OPERATION_CODES.put("QUOTA_EXCEEDED", 400);
+	}
+
+	private GoogleOperationFailedException(@Nonnegative int httpCode, @Nullable String providerCode, @Nonnull String msg) {
 		super(CloudErrorType.GENERAL, httpCode, providerCode, msg);
 	}
 
-	public GoogleOperationFailedException(@Nonnegative int httpCode, @Nullable String providerCode, @Nonnull String msg, @Nonnull Throwable cause) {
+	private GoogleOperationFailedException(@Nonnegative int httpCode, @Nullable String providerCode, @Nonnull String msg, @Nonnull Throwable cause) {
 		super(CloudErrorType.GENERAL, httpCode, providerCode, msg, cause);
 	}
 
 	public static GoogleOperationFailedException create(Operation operation) {
-		List<Operation.Error.Errors> errors = operation.getError().getErrors();
+		List<Errors> errors = operation.getError().getErrors();
 		return create(operation, errors.get(0).getMessage());
 	}
 
 	public static GoogleOperationFailedException create(Operation operation, String errorMessage) {
-		return new GoogleOperationFailedException(operation.getHttpErrorStatusCode(), operation.getHttpErrorMessage(), errorMessage);
+		List<Errors> errors = operation.getError().getErrors();
+
+		Errors firstError = errors.get(0);
+		int httpStatusCode = OPERATION_CODES.containsKey(firstError.getCode())
+				? OPERATION_CODES.get(firstError.getCode()) : operation.getHttpErrorStatusCode();
+
+		return new GoogleOperationFailedException(httpStatusCode, operation.getHttpErrorMessage(), errorMessage);
 	}
+
 }

